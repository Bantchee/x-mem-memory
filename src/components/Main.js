import React, {useState} from "react";
import Card from "./Card";
import ScoreContainer from "./ScoreContainer";

const Main = (props) => {
    const [currentScore, setCurrentScore] = useState(0);
    const [bestScore, setBestScore] = useState(0);
    const [cards, setCards] = useState(
        [
            {
                name: "Wolverine",
                imgUrl: "../img/wolverine.jpeg",
                hasBeenClicked: false,
            },
            {
                name: "Cyclops",
                imgUrl: "../img/cyclops.jpeg",
                hasBeenClicked: false,
            },
            {
                name: "Professor X",
                imgUrl: "../img/professor_x.jpeg",
                hasBeenClicked: false,
            },
            {
                name: "Storm",
                imgUrl: "../img/storm.jpeg",
                hasBeenClicked: false,
            },
            {
                name: "Jean Grey",
                imgUrl: "../img/jean_grey.jpeg",
                hasBeenClicked: false,
            },
            {
                name: "Nightcrawler",
                imgUrl: "../img/nightcrawler.jpeg",
                hasBeenClicked: false,
            },
            {
                name: "Gambit",
                imgUrl: "../img/gambit.jpeg",
                hasBeenClicked: false,
            },
            {
                name: "Rogue",
                imgUrl: "../img/rogue.jpeg",
                hasBeenClicked: false,
            },
            {
                name: "Colossus",
                imgUrl: "../img/colossus.jpeg",
                hasBeenClicked: false,
            },
            {
                name: "Beast",
                imgUrl: "../img/beast.jpeg",
                hasBeenClicked: false,
            },
            {
                name: "Angel",
                imgUrl: "../img/angel.jpeg",
                hasBeenClicked: false,
            },
            {
                name: "Iceman",
                imgUrl: "../img/iceman.jpeg",
                hasBeenClicked: false,
            },
        ]
    );

    const renderCard = (index, name, imgUrl) => {
        return (
            <Card 
                key={index}
                name={name}
                imgUrl={imgUrl}
            />
        )
    };

    const shuffleCards = (cards) => {
        return cards;
    };

    return (
        <div
            className="row-span-5 bg-neutral-900 p-4 flex flex-col gap-4"
        >
            <ScoreContainer
                currentScore={currentScore}
                bestScore={bestScore}
            />
            <div
                className="flex flex-wrap gap-4 justify-center pr-96 pl-96"
            >
                {
                    cards.map(
                        (card, index) => {
                            return renderCard(index, card.name, card.imgUrl);  
                        }
                    )
                }
            </div>
            
        </div>
    );
};

export default Main;