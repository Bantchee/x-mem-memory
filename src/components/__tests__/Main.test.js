import Main from "../Main";
import { render, screen, act } from "@testing-library/react"
import "@testing-library/jest-dom"
import userEvent from "@testing-library/user-event";
// import { act } from "react-dom/test-utils";

describe("Main Component", () => {
    test("renders", () => {
        const { container } = render(<Main />);
        expect(container).toMatchSnapshot();
    });


});

describe("Card Clicked Once", () => {
    test("Current Score increases", () => {
        render(<Main />);
        const cards = screen.getAllByRole("button");
        const currentScore = screen.getByText(/Current Score:/i);

        userEvent.click(cards[0]);
        expect(currentScore.textContent).toMatch(/Current Score: 1/i);
    });

    test("Best Score increases", () => {
        render(<Main />);
        const cards = screen.getAllByRole("button");
        const bestScore = screen.getByText(/Best Score:/i);

        userEvent.click(cards[0]);
        
        expect(bestScore.textContent).toMatch(/Best Score: 1/i);
    });
});