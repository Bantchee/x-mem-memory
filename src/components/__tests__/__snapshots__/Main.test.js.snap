// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Component renders 1`] = `
<div>
  <div
    class="row-span-5 bg-neutral-900 p-4 flex flex-col gap-4"
  >
    <div
      class="flex text-2xl gap-16 justify-center"
    >
      <div
        class="pt-1 pb-1 border-4 w-64 flex justify-center rounded-lg border-yellow-400"
      >
        <span
          class="text-red-500 m-auto"
        >
          Current Score: 
          0
        </span>
      </div>
      <div
        class="pt-1 pb-1 border-4 w-64 flex justify-center rounded-lg border-yellow-400"
      >
        <span
          class="text-red-500 m-auto"
        >
          Best Score: 
          0
        </span>
      </div>
    </div>
    <div
      class="flex flex-wrap gap-4 justify-center pr-96 pl-96"
    >
      <button
        class="flex flex-col content-center justify-center border-red-500 border-4 rounded-xl p-2 bg-yellow-400"
      >
        <img
          alt="Wolverine"
          class="rounded-xl w-28 h-36"
          src="wolverine.jpeg"
        />
        <p
          class="text-neutral-900 bottom-0 text-center font-bold text-base"
        >
          Wolverine
        </p>
      </button>
      <button
        class="flex flex-col content-center justify-center border-red-500 border-4 rounded-xl p-2 bg-yellow-400"
      >
        <img
          alt="Cyclops"
          class="rounded-xl w-28 h-36"
          src="cyclops.jpeg"
        />
        <p
          class="text-neutral-900 bottom-0 text-center font-bold text-base"
        >
          Cyclops
        </p>
      </button>
      <button
        class="flex flex-col content-center justify-center border-red-500 border-4 rounded-xl p-2 bg-yellow-400"
      >
        <img
          alt="Professor X"
          class="rounded-xl w-28 h-36"
          src="professor_x.jpeg"
        />
        <p
          class="text-neutral-900 bottom-0 text-center font-bold text-base"
        >
          Professor X
        </p>
      </button>
      <button
        class="flex flex-col content-center justify-center border-red-500 border-4 rounded-xl p-2 bg-yellow-400"
      >
        <img
          alt="Storm"
          class="rounded-xl w-28 h-36"
          src="storm.jpeg"
        />
        <p
          class="text-neutral-900 bottom-0 text-center font-bold text-base"
        >
          Storm
        </p>
      </button>
      <button
        class="flex flex-col content-center justify-center border-red-500 border-4 rounded-xl p-2 bg-yellow-400"
      >
        <img
          alt="Jean Grey"
          class="rounded-xl w-28 h-36"
          src="jean_grey.jpeg"
        />
        <p
          class="text-neutral-900 bottom-0 text-center font-bold text-base"
        >
          Jean Grey
        </p>
      </button>
      <button
        class="flex flex-col content-center justify-center border-red-500 border-4 rounded-xl p-2 bg-yellow-400"
      >
        <img
          alt="Nightcrawler"
          class="rounded-xl w-28 h-36"
          src="nightcrawler.jpeg"
        />
        <p
          class="text-neutral-900 bottom-0 text-center font-bold text-base"
        >
          Nightcrawler
        </p>
      </button>
      <button
        class="flex flex-col content-center justify-center border-red-500 border-4 rounded-xl p-2 bg-yellow-400"
      >
        <img
          alt="Gambit"
          class="rounded-xl w-28 h-36"
          src="gambit.jpeg"
        />
        <p
          class="text-neutral-900 bottom-0 text-center font-bold text-base"
        >
          Gambit
        </p>
      </button>
      <button
        class="flex flex-col content-center justify-center border-red-500 border-4 rounded-xl p-2 bg-yellow-400"
      >
        <img
          alt="Rogue"
          class="rounded-xl w-28 h-36"
          src="rogue.jpeg"
        />
        <p
          class="text-neutral-900 bottom-0 text-center font-bold text-base"
        >
          Rogue
        </p>
      </button>
      <button
        class="flex flex-col content-center justify-center border-red-500 border-4 rounded-xl p-2 bg-yellow-400"
      >
        <img
          alt="Colossus"
          class="rounded-xl w-28 h-36"
          src="colossus.jpeg"
        />
        <p
          class="text-neutral-900 bottom-0 text-center font-bold text-base"
        >
          Colossus
        </p>
      </button>
      <button
        class="flex flex-col content-center justify-center border-red-500 border-4 rounded-xl p-2 bg-yellow-400"
      >
        <img
          alt="Beast"
          class="rounded-xl w-28 h-36"
          src="beast.jpeg"
        />
        <p
          class="text-neutral-900 bottom-0 text-center font-bold text-base"
        >
          Beast
        </p>
      </button>
      <button
        class="flex flex-col content-center justify-center border-red-500 border-4 rounded-xl p-2 bg-yellow-400"
      >
        <img
          alt="Angel"
          class="rounded-xl w-28 h-36"
          src="angel.jpeg"
        />
        <p
          class="text-neutral-900 bottom-0 text-center font-bold text-base"
        >
          Angel
        </p>
      </button>
      <button
        class="flex flex-col content-center justify-center border-red-500 border-4 rounded-xl p-2 bg-yellow-400"
      >
        <img
          alt="Iceman"
          class="rounded-xl w-28 h-36"
          src="iceman.jpeg"
        />
        <p
          class="text-neutral-900 bottom-0 text-center font-bold text-base"
        >
          Iceman
        </p>
      </button>
    </div>
  </div>
</div>
`;
